src/sentry/replays/consumers/recording.py

```diff
@@ -188,14 +201,34 @@ def parse_headers(recording: bytes, replay_id: str) -> tuple[int, bytes]:
 # I/O Task
 
 
-def commit_message_with_options(message: Message[ProcessedEvent]) -> None:
-    profiling_enabled = options.get(
-        "replay.consumer.recording.profiling.enabled",
+def commit_message_with_profiling(message: Message[ProcessedEvent]) -> None:
+    # Only initialize profiling if we have a DSN and sample rate > 0
+    profiling_dsn = getattr(
+        settings, "SENTRY_REPLAY_RECORDINGS_CONSUMER_PROFILING_PROJECT_DSN", None
+    )
+    profile_session_sample_rate = getattr(
+        settings, "SENTRY_REPLAY_RECORDINGS_CONSUMER_PROFILING_SAMPLE_RATE", 0
     )
-    return commit_message(message, profiling_enabled=profiling_enabled)
+    profiling_active = profiling_dsn is not None and profile_session_sample_rate > 0
```

Comment:
If this causes issues our only recourse is to revert the getsentry PR.  That will take many hours.  We need an option which can toggle it on and off quickly.